### PR TITLE
Add a how-to guide section

### LINKED ISSUE #5427: Add a "How do I..." page to our documentation
e.g. this page from the xarray docs: http://xarray.pydata.org/en/stable/howdoi.html

### LINKED ISSUE #6909: Add a "How-to Guide" section to the docs
### Describe the feature

In the vein of refactoring our documentation in terms of the diátaxis framework, our documentation should have a ["how-to guide" section](https://diataxis.fr/how-to-guides/). 

The [community meeting notes](https://github.com/sunpy/sunpy/wiki/Home%3A-Weekly-Community-Meetings-Archive#5-april-2023) have some useful discussion regarding how we might go about this. We have a lot of material, in the example gallery and in some of the API docs, that could be place (and some cases reformatted) as how-to guides. When it comes to populating the how-to guide, a good place to start is probably the example gallery and to assess which entries would be better recast as how-to guides.

### Proposed solution

_No response_

### PR DESCRIPTION
**Pull request by wtbarnes**
**Please rebase and merge. Do not squash merge.**

Fixes #6909 
Fixes #5427 

I've converted one entry from the example gallery to a how-to guide which I thought was a fairly uncontroversial example. I'd like to limit this PR to just adding the section/general structure and then leave a larger restructuring of other content to how-to guides to other PRs. A debate on what in the example gallery should be moved here would tie this up indefinitely. 

I'd mostly like to decide on naming, formatting, high-level text and general wording for the descriptions of the sections here. Once we have this merged, we can start reorganizing more asynchronously.

### CODE REVIEW

File: docs/how_to/index.rst [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+.. list-table::
+   :widths: auto
+   :header-rows: 1
+
+   * - How do I...
+     - Solution
+   * - create a map from a FITS file
+     - :mod:`Map('file.fits') <sunpy.map>`
+   * - save a map to a FITS file
+     - :meth:`Map.save() <sunpy.map.GenericMap.save>`
```
Comment (nabobalis) [inline docs/how_to/index.rst:32]: I thought we didn't need the `()` if we add `:meth:`?

File: docs/index.rst [hunk 2]
```diff
@@ -33,19 +33,10 @@ Once you've installed sunpy, start here.
...
-    When a page or subgallery containing more diataxis like how-to (longer form) guides is added then link here
-    [How to guides] provide detailed overview on some of the features of sunpy. These assume some level of experience using sunpy.
-
-.. todo::
-
-    Index explanation pages (e.g. different map rotation methods and then link here.)
-    [Topic guides] discuss key topics and concepts at a fairly high level and provide useful background information and explanation.
+* :ref:`how_to_guide` contains short snippets of code for accomplishing specific tasks with sunpy. This section is most useful for answering "How do I..." questions.
+* :ref:`guide` contains in-depth explanations of key topics and concepts in sunpy. This section of the documentation is more advanced than the others and is most useful for answering "why" questions.
+* :ref:`API reference<reference>` contains a technical description of the inputs, outputs, and behavior of each component of sunpy core.
```
Comment (nabobalis) [inline docs/index.rst:39]:
```suggestion
* :ref:`API reference <reference>` contains a technical description of the inputs, outputs, and behavior of each component of sunpy core.
```

**Review by nabobalis — APPROVED** [review]

File: docs/how_to/index.rst (hunk 1 above)
Comment (nabobalis) [reply docs/how_to/index.rst:32]: Then should we remove `:meth:`?

Comment (wtbarnes) [reply docs/how_to/index.rst:32]: Yeah I'll just remove them.

File: docs/how_to/index.rst [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+For more complete code examples, see the how-to guides above.
+
+.. list-table::
+   :widths: auto
+   :header-rows: 1
+
+   * - How do I...
+     - Solution
+   * - create a map from a FITS file
+     - :mod:`Map('file.fits') <sunpy.map>`
```
Comment (dstansby) [inline docs/how_to/index.rst:30]:
```suggestion
     -``my_map=`` `Map <sunpy.map.Map>` ``('my_file.fits')
```

File: docs/how_to/index.rst [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+   * - save a map to a FITS file
+     - `Map.save() <sunpy.map.GenericMap.save>`
```
Comment (dstansby) [inline docs/how_to/index.rst:32]:
```suggestion
     - `my_map.save() <sunpy.map.GenericMap.save>`
```

**Review by dstansby — CHANGES_REQUESTED** [review]
Two suggestions:

For the map stuff, I've left inline comments. Essentially I think having code that loads a map into a variable at the top, then using that variable in the code snippets below would be a bit clearer.

I think some words could be saved by splitting the table into different sections. For e.g. the map section, there could be a "Maps" title at the top, and then the text could change like:
- "create a map from a FITS file" > "load from a FITS file"
- "save a map to a FITS file" > "save to a FITS file"
- "get a quicklook summary of a map" > "get a quicklook"

removing all the "a map" bits would make it easier to read IMO

**Comment by wtbarnes** [comment]
> Essentially I think having code that loads a map into a variable at the top, then using that variable in the code snippets below would be a bit clearer.

I initially avoided this because I was directly linking to the docstring, but since I'm constructing the links manually anyway, I think this does make more sense. 

> I think some words could be saved by splitting the table into different sections. 

I disagree. I think the table should be as simple as possible and as short as possible, particularly since these code snippets are not tested. I'm worried that if it starts increasing in complexity and length, we'll just be duplicating the API docs and the table will lose its purpose.

**Comment by wtbarnes** [comment]
> 👍 - I still definietely thing there are words to be cut down for brevity/readability, but I'm happy to bikeshed that in a future PR instead of here.

Agreed. I made a pass at making it a bit less wordy and avoiding repetition.

**Review by nabobalis — APPROVED** [review]

**Review by dstansby — APPROVED** [review]
👏

**Comment by wtbarnes** [comment]
🎉  We now have spaces for all four diataxis sections!

**Pull request merged**